File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    let n = x;
    let tmp = 0;
    while(n) {
        tmp += n % 10;
        n = parseInt(n / 10);
    };
    if (x % tmp == 0) {
        return true;
    } else {
        return false;
    }
}

File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int hashard = 0;
    int y = x;
    while(y) {
        hashard += y % 10;
        y = int(y / 10);
    };
    return !(x % hashard);
}